### PR TITLE
[XC] import types in StaticResourceExtension

### LINKED ISSUE #25871: XAMLC error when building an app with AoT compilation
### Description

When building an app using AoT in release mode it breaks during build with the error message:

> C:\Users\userName\.nuget\packages\microsoft.maui.controls.build.tasks\9.0.10\buildTransitive\netstandard2.0\Microsoft.Maui.Controls.targets(171,3): Member 'System.IComparable' is declared in another module and needs to be imported [\MCT\samples\CommunityToolkit.Maui.Sample\CommunityToolkit.Maui.Sample.csproj]

### Steps to Reproduce

1. Run the sample project on the toolkit [repo](https://github.com/CommunityToolkit/Maui/tree/feature/sl-dotnet-nine) (be sure to that you're on `feature/sl-dotnet-nine` branch
2. open the terminal and go to `\samples\CommunityToolkit.Maui.Sample` 
3. run, on terminal, `dotnet build -c Release -bl`
4. See the message error

Binlog:
[msbuild.zip](https://github.com/user-attachments/files/17758211/msbuild.zip)

### Link to public reproduction project repository

https://github.com/CommunityToolkit/Maui/tree/feature/sl-dotnet-nine

### Version with bug

9.0.10 SR1

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.100 SR10

### Affected platforms

iOS, Android, Windows, macOS

### Affected platform versions

All versions

### Did you find any workaround?

N/A

### Relevant log output

```shell

```

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change

compiled StaticResource is new to net9.0. there was a case when the a external type wasn't imported. fixing this

### Issues Fixed

- fixes #25871

### CODE REVIEW

File: src/Controls/src/Build.Tasks/SetPropertiesVisitor.cs [hunk 1]
```diff
@@ -1540,7 +1540,7 @@ static IEnumerable<Instruction> Set(VariableDefinition parent, string localName,
 			module.ImportReference(parent.VariableType.ResolveCached(context.Cache));
 			var propertySetterRef = module.ImportReference(module.ImportReference(propertySetter).ResolveGenericParameters(declaringTypeReference, module));
 			propertySetterRef.ImportTypes(module);
-			var propertyType = property.PropertyType.ResolveGenericParameters(declaringTypeReference);
+			var propertyType = module.ImportReference(property.PropertyType.ResolveGenericParameters(declaringTypeReference));
```
Comment (StephaneDelcroix) [inline src/Controls/src/Build.Tasks/SetPropertiesVisitor.cs:1543]: not really required for this issue, but might be for others

**Comment by bijington** [comment]
@StephaneDelcroix thanks for this. I was just looking at making a change myself at https://github.com/bijington/maui-1/tree/feature/sl-xamlc-icomparable but then you are more likely to know how to fix this over me.

**Review by jfversluis — APPROVED** [review]

**Comment by bijington** [comment]
I just wanted to say thanks again for this fix. I have run the community toolkit build with the nightly version of `9.0.20-ci.main.24604.4` and I can confirm this has resolved our issue on Android.

**Comment by MauiUIui** [comment]
Thanks guys :)

**Pull request merged**